### ISSUE TITLE
Broken cursor and current line highligting after CtrlSpace

### ISSUE THREAD

**Issue by unclechu** [extr0py/oni#401]
When I enter `CtrlSpace` window and close it I have my cursor looking as in *insert-mode* and current line isn't highlighted until I go to different mode and then back, or for example press `%` (move around brackets).

Minimal `init.vim` to reproduce it:

```viml
filetype off
exec 'set rtp+='. ($HOME) .'/.config/nvim/bundle/Vundle.vim'                                                                                             
call vundle#begin($HOME . '/.config/nvim/bundle')                                                                                                        
Plugin 'gmarik/Vundle.vim', {'pinned': 1} " provided by git-submodule                                                                                    
Plugin 'vim-ctrlspace/vim-ctrlspace'                                                                                                                                
call vundle#end()                                                                                                                                        
filetype plugin indent on 
set hidden
```

Short screencast that demonstrates the issue: https://youtu.be/UgTj4i49T5Y

I'm sure it isn't connected directly to the CtrlSpace plugin but the Oni's issue, it was just obvious way for me to reproduce it. In `nvim` TUI or `nvim-qt` GUI it can't be reproduced.

**Comment by keforbes** [comment]
This is probably related to #331.

**Comment by extr0py** [comment]
Yep, seems like the same issue as #331 - just a different manifestation of it. Thanks for including the video, @unclechu , that's very helpful.

**Comment by unclechu** [comment]
I built Neovim 0.2.0 release (without updating Oni) and now current line is highlighted after I leave CtrlSpace window but cursor is still thin line in *normal-mode* until I reset it.

**Comment by extr0py** [comment]
This should be addressed by #530 - please reopen if that fix does not address this.
